apps/sms/js/navigation.js

```diff
@@ -116,6 +116,17 @@ var Navigation = {
   },
 
   /**
+   * Ensures that current panel is correctly set, if no current panel set and
+   * we're not in the process of transitioning then set default panel as current
+   * one.
+   */
+  ensureCurrentPanel: function() {
```

Comment:
Okay, will go with ``` ensureCurrentPanel``` for now. I also have some things in mind similar to what  you've described, so we're good :)